server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -114,6 +118,22 @@ public AllocationService(
         this.shardsAllocator = shardsAllocator;
         this.clusterInfoService = clusterInfoService;
         this.snapshotsInfoService = snapshotsInfoService;
+        this.settings = Settings.EMPTY;
```

Comment:
We can use constructor chaining for Line 117 - 121:

```
this(allocationDeciders, 
    shardsAllocator, 
    clusterInfoService, 
    snapshotsInfoService, 
    Settings.EMPTY);
```

Reply:
Ack